dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java

```diff
@@ -1064,95 +729,133 @@ private AppsSecretsImportExport collectSecretsForExport(final Map<String, Set<St
+                            .format("No site identified by `%s` was found locally.", siteId));
+                    continue;
+                }
+                throw new IllegalArgumentException(
+                        String.format("No site identified by `%s` was found locally.\n %s", siteId,
+                                failSilentlyMessage));
+            }
+
+            for (final AppSecrets appSecrets : importEntry.getValue()) {
+                final Optional<AppDescriptor> appDescriptor = getAppDescriptor(appSecrets.getKey(),
+                        user);
+                if (!appDescriptor.isPresent()) {
+                    if (failSilently) {
+                        Logger.warn(AppsAPIImpl.class, () -> String
+                                .format("No App Descriptor `%s` was found locally.",
+                                        appSecrets.getKey()));
+                        continue;
+                    }
+                    throw new IllegalArgumentException(
+                            String.format("No App Descriptor `%s` was found locally.\n %s",
+                                    appSecrets.getKey(), failSilentlyMessage));
+                }
+
+                if (appSecrets.getSecrets().isEmpty()) {
+                    if (failSilently) {
+                        Logger.warn(AppsAPIImpl.class, () -> String
+                                .format("Incoming empty secret `%s` will be skipped.",
+                                        appSecrets.getKey()));
+                        continue;
+                    }
+                    throw new IllegalArgumentException(
+                            String.format("Incoming empty secret `%s` could replace local copy.\n %s ",
+                                    appSecrets.getKey(), failSilentlyMessage));
+                }
+                try {
+                    validateForSave(mapForValidation(appSecrets), appDescriptor.get());
+                } catch (IllegalArgumentException ae) {
+                    if (failSilently) {
+                        Logger.warn(AppsAPIImpl.class, () -> String
+                                .format("Incoming secret `%s` has validation issues with local descriptor will be skipped.",
+                                        appSecrets.getKey()));
+                        continue;
+                    }
+                    throw new IllegalArgumentException(
+                            String.format(
+                                    "Incoming secret `%s` has validation issues with local descriptor.\n %s ",
+                                    appSecrets.getKey(), failSilentlyMessage),
+                            ae);
+                }
+                Logger.info(AppsAPIImpl.class, String.format("Imported secret `%s` ", appSecrets));
+                saveSecrets(appSecrets, site, user);
+                count++;
+            }
+        }
+        if(count >= 1){
```

Comment:
Codacy found an issue: [Avoid using Literals in Conditional Statements](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6286472)